doc/man/pilot-install-user.1
.\"Generated by db2man.xsl. Don't modify this, modify the source.
.de Sh \" Subsection
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Ip \" List item
.br
.ie \\n(.$>=3 .ne \\$3
.el .ne 3
.IP "\\$1" \\$2
..
.TH "pilot-install-user" 1 "Copyright 1996-2007 FSF" "0.12.5" "PILOT-LINK"
.SH NAME
pilot-install-user \- Reads or sets a Palm User and UserID on a Palm Device.
.SH "SECTION"

.PP
pilot\-link: Userland conduits

.SH "SYNOPSIS"

.PP
 \fIpilot\-install\-user\fR [\fB\-p\fR|\fB\-\-port\fR <\fBport\fR>] [\fB\-\-version\fR] [\fB\-?\fR|\fB\-\-help\fR] [\fB\-\-usage\fR] [\fB\-q\fR|\fB\-\-quiet\fR] [\fB\-u\fR|\fB\-\-user\fR  \fBusername\fR] [\fB\-i\fR|\fB\-\-id\fR  \fBuserid\fR] [\fB\-l\fR|\fB\-\-list\fR]

.SH "DESCRIPTION"

.PP
 \fIpilot\-install\-user\fR allows the user to read or change the User and UserID information on the Palm handheld\&.

.SH "OPTIONS"

.SS "pilot-install-user options"

                        \fB\-u\fR,
                        \fB\-\-user\fR \fBusername\fR
                    
.PP
Set the username, use quotes for spaces in the username, as in "John Q\&. Public"\&.

                        \fB\-i\fR,
                        \fB\-\-id\fR \fBuserid\fR
                    
.PP
Set the 5\-digit numeric UserID on the device (used for many Palm application registration methods)\&.

                        \fB\-l\fR, \fB\-\-list\fR
                    
.PP
List the current username and ID (default)

.SS "Conduit Options"

                        \fB\-p\fR, \fB\-\-port\fR
                        <\fBport\fR>
                    
.PP
Use device file <\fIport\fR> to communicate with the Palm handheld\&. If this is not specified, \fIpilot\-install\-user\fR will look for the \fI $PILOTPORT \fR environment variable\&. If neither are found or supplied, \fIpilot\-install\-user\fR will print the usage information\&.

                        \fB\-q\fR, 
                        \fB\-\-quiet\fR
                    
.PP
Suppress 'Hit HotSync button' message

                        \fB\-v\fR, \fB\-\-version\fR
                    
.PP
Display version of \fIpilot\-install\-user\fR and exit without connecting\&.

.SS "Help Options"

                        \fB\-h\fR, \fB\-\-help\fR
                    
.PP
Display the help synopsis for \fIpilot\-install\-user\fR and exit without connecting\&.

                        \fB\-\-usage\fR 
                    
.PP
Display a brief usage message and exit without connecting\&.

.SH "USAGE"

.PP
The program will connect to a target Palm handheld, and read or set the Palm UserName specified or display the Palm username and UserID if none is specified\&.

.SH "AUTHOR"

.PP
 \fIpilot\-install\-user\fR was originally written by D\&. Jeff Dionne <jeff@RyeHam\&.ee\&.ryerson\&.ca>\&.

.PP
This manual page was converted to XML by Neil Williams <linux@codehelp\&.co\&.uk> 

.SH "REPORTING BUGS"

.PP
We have an online bug tracker\&. Using this is the only way to ensure that your bugs are recorded and that we can track them until they are resolved or closed\&. Reporting bugs via email, while easy, is not very useful in terms of accountability\&. Please point your browser to http://bugs\&.pilot\-link\&.org: \fIhttp://bugs.pilot-link.org\fR and report your bugs and issues there\&.

.SH "COPYRIGHT"

.PP
This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2 of the License, or (at your option) any later version\&.

.PP
This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE\&. See the GNU General Public License for more details\&.

.PP
You should have received a copy of the GNU General Public License along with this program; if not, write to the Free Software Foundation, Inc\&., 51 Franklin St, Fifth Floor, Boston, MA 02110\-1301, USA\&.

.SH "SEE ALSO"

.PP
 \fIpilot\-install\-netsync\fR (1), \fIpilot\-dlpsh\fR (1), \fIpilot\-link\fR(7)\&.



src/pilot-install-user.c
/*
 * $Id: pilot-install-user.c,v 1.68 2006/10/12 14:21:21 desrod Exp $ 
 *
 * install-user.c:  Palm Username installer
 *
 * This program is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the
 * Free Software Foundation; either version 2 of the License, or (at your
 * option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
 * Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 *
 */

#include <stdio.h>

#include "pi-dlp.h"
#include "pi-header.h"
#include "pi-userland.h"

const char
    *user   = NULL,
    *userid = NULL;
int list = 0;

const struct poptOption options[] = {
	USERLAND_RESERVED_OPTIONS
	{ "user",    'u', POPT_ARG_STRING, &user, 0, "Set the username, use quotes for spaces (see example)", "<username>"},
	{ "id",      'i', POPT_ARG_STRING, &userid, 0, "A 5-digit numeric UserID, required for PalmOS", "<userid>"},
	{ "list",    'l', POPT_ARG_NONE, &list, 0, "List the current username and ID (default)",NULL},
	POPT_TABLEEND
};

poptContext po;

int main(int argc, char *argv[])
{
	int  sd     = -1,
	     po_err = -1;

        char    *environment;
	struct 	PilotUser 	User;

        environment = getenv("PILOTPORT");

        po = poptGetContext("install-user", argc, (const char **) argv, options, 0);
	poptSetOtherOptionHelp(po, "\n\n"
		"   Set a  username and ID on the Palm.\n\n"
		"   Example:\n"
		"      -p /dev/pilot -u \"John Q. Public\" -i 12345\n");

	if ((environment == NULL) && (argc < 2)) {
		poptPrintUsage(po,stderr,0);
		return 1;
	}

        while ((po_err = poptGetNextOpt(po)) >= 0) {
		/* Everything is handled by popt magic */
	}

	if (po_err < -1) {
		plu_badoption(po,po_err);
	}

	sd = plu_connect();
	if (sd < 0) {
		goto error;
	}

	if (dlp_ReadUserInfo(sd, &User) < 0) {
		goto error_close;
	}

	if (list || (!user && !userid)) {
		printf("   Palm user: %s\n", User.username);
		printf("   UserID:    %lu\n\n", User.userID);
		goto cleanup;
	}

	if (userid) {
		User.userID = abs(atoi(userid));
	}

	if (user) {
		strncpy(User.username, user, sizeof(User.username) - 1);
	}

	User.lastSyncDate = time(NULL);

	if (dlp_WriteUserInfo(sd, &User) < 0) {
		goto error_close;
	}

	if (!plu_quiet && user) {
		printf("   Installed User Name: %s\n", User.username);
	}
	if (!plu_quiet && userid) {
		printf("   Installed User ID: %lu\n", User.userID);
	}
	if (!plu_quiet && (user || userid)) {
		printf("\n");
	}

cleanup:
	if (dlp_AddSyncLogEntry(sd, "install-user exited normally.\n"
				    "Thank you for using pilot-link.\n") < 0) {
		goto error_close;
	}

	if (dlp_EndOfSync(sd, 0) < 0) {
		goto error_close;
	}

	if (pi_close(sd) < 0) {
		goto error;
	}

	return 0;

error_close:
	pi_close(sd);

error:
	return -1;
}

/* vi: set ts=8 sw=4 sts=4 noexpandtab: cin */
/* ex: set tabstop=4 expandtab: */
/* Local Variables: */
/* indent-tabs-mode: t */
/* c-basic-offset: 8 */
/* End: */

